lib/src/consts.rs

```diff
@@ -122,16 +125,51 @@ lazy_static! {
     };
 }
 
-pub fn get_network_spec(network: Network) -> ChainSpec {
-    match network {
-        Network::Ethereum => ETH_MAINNET_CHAIN_SPEC.clone(),
-        Network::Holesky => ETH_HOLESKY_CHAIN_SPEC.clone(),
-        Network::TaikoA7 => TAIKO_A7_CHAIN_SPEC.clone(),
+#[derive(Clone, Debug)]
+pub struct SupportedChainSpecs(HashMap<String, ChainSpec>);
+
+impl SupportedChainSpecs {
+    pub fn default() -> Self {
+        SupportedChainSpecs(
+            [
+                ETH_MAINNET_CHAIN_SPEC.clone(),
+                ETH_HOLESKY_CHAIN_SPEC.clone(),
+                TAIKO_A7_CHAIN_SPEC.clone(),
+            ]
+            .into_iter()
+            .map(|cs| (cs.name.clone(), cs))
+            .collect(),
+        )
+    }
+
+    #[cfg(feature = "std")]
+    pub fn merge_from_file(file_path: PathBuf) -> Result<SupportedChainSpecs> {
```

Comment:
If we just always just load the chain spec directly from disk then users can customize the chain spec there however they want, do we need this merging? Because then "merging" is left to the users using whatever default configs we provide and they just override the value there (or create a new config file from it). Would that do everything we need?

So basically we don't have to care about any "supported" chains that way I think. The user just specifies a path, we load the config as is, and then we use that. So by default we provide `ethereum.json`, `holesky.json`, and `taiko_a7.json`, and then users could either use/modify those however they want or they can create a new `custom_chain.json` and use that.

Reply:
I thought that before. now the default is just for those who don't want to customize sth, but all defaults can be override by conf if the name is the same. Also I thought no need to ask existed tests to read files, as most customized network should not has unit test requirements. But yes, we can change to pure input conf dependency.

Reply:
Ah I see. Current approach is fine with me, I do think just loading always from file would be fine and makes things a bit simpler. But otherwise no hard preference.

Reply:
Oh we could even do `include_str!("chain_spec.json")` to inject the config file into the rust code at compile time, that way we can even avoid reading anything from a file in the normal case at runtime and we don't need to explicitly have these configs in the code. Don't know if that's worth it, just an idea. Because now the chain config data may be duplicated between the in code version and the config file.

Reply:
That seems make sense, let me think of it. Basically it relies on a pre-built const default conf file, but it still needs a overriding chain_spec_file cmd param?? because I think we need dynamic conf (runtime behavior) rather than re-compiling in each time conf changes. 
how about we change current default.json to default.template.json? As the chain_spec is a optional param, we don't really need to provide a default.json conf, so (somehow) no duplication in code & file, just a show case of format.

Reply:
So we would have the config there with all the default values in the json as part of the project (like the other config files), which will get compiled in. If people want to use custom changes (without recompiling) they can just modify the standard json file in the project (or of course create a new one) and set the path to that config file as input so raiko will load that one in dynamically yeah.

> how about we change current default.json to default.template.json? As the chain_spec is a optional param, we don't really need to provide a default.json conf, so (somehow) no duplication in code & file, just a show case of format.

Also works, though then we'll have to make sure the `default.template.json` is kept up to date/tested specifically because if the file is not part of the normal flow things tend to break.

Reply:
changed to include_str!, just one more thought that now the normal path in lib involves serde (maybe previously it also is), maybe some side effects in VM side? (just guessing, as un-touched code should be removed by the compiler, not familiar with that).